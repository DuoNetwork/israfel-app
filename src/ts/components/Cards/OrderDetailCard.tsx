import close from 'images/icons/close.svg';
import * as React from 'react';
import * as CST from 'ts/common/constants';
import { IUserOrder } from 'ts/common/types';
import util from 'ts/common/util';
import { SCard, SCardList, SCardTitle } from './_styled';

interface IProps {
	orders: IUserOrder[];
	handleClose: () => void;
}

export default class OrderDetailCard extends React.Component<IProps> {
	public render() {
		const { orders, handleClose } = this.props;
		const visible = orders.length > 0;
		if (!visible) return null;
		const lastVersion = orders[0];
		return (
			<div style={{ display: visible ? 'block' : 'none' }}>
				<div className={'popup-bg ' + (visible ? 'popup-open-bg' : '')} />
				<SCard
					title={
						<SCardTitle>
							{CST.TH_ORDER + ' ' + CST.TH_DETAIL}
						</SCardTitle>
					}
					style={{ overflow: 'visible !important' }}
					width="480px"
					className={'popup-card-oh ' + (visible ? 'popup-open' : '')}
					noBodymargin
					extra={<img className="cardpopup-close" src={close} onClick={handleClose} />}
				>
					<SCardList noMargin width="100%">
						<div className="status-list-wrapper">
							<ul style={{ padding: '0 10px', maxHeight: 256, overflowY: 'scroll' }}>
								<li>{util.getOrderDescription(lastVersion)}</li>
								{orders.map(o => (
									<li
										key={o.currentSequence}
										onClick={() => {
											if (o.transactionHash)
												window.open(
													`https://${
														__ENV__ === CST.DB_LIVE ? '' : 'kovan.'
													}etherscan.io/tx/${o.transactionHash}`,
													'_blank'
												);
										}}
										style={{cursor: o.transactionHash ? 'pointer' : 'default'}}
									>
										<span>
											{util.formatTimeSecond(o.updatedAt || o.createdAt)}
										</span>
										<span>{util.getVersionDescription(o)}</span>
									</li>
								))}
							</ul>
						</div>
					</SCardList>
				</SCard>
			</div>
		);
	}
}
