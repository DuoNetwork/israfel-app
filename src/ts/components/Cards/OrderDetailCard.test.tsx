/**
 * @jest-environment jsdom
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import OrderDetailCard from './OrderDetailCard';

describe('OrderDetailCard Test', () => {
	describe('Test Snapshot', () => {
		const orders = [
			{
				type: 'test',
				status: 'test',
				updatedBy: 'test',
				processed: true,
				account: 'test',
				pair: 'test',
				orderHash: 'test',
				price: 123,
				amount: 123,
				balance: 123,
				matching: 123,
				fill: 123,
				side: 'test',
				expiry: 123,
				createdAt: 123,
				updatedAt: 123,
				initialSequence: 123,
				currentSequence: 123,
				fee: 123,
				feeAsset: 'test'
			}
		];
		const handleClose = jest.fn(() => ({}));
		it('Test Snapshot', () => {
			const wrapper = shallow(<OrderDetailCard orders={orders} handleClose={handleClose} />);
			expect(wrapper).toMatchSnapshot();
		});
	});
});
