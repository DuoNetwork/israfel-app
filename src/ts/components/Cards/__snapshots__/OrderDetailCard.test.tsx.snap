// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailCard Test Test Snapshot Test Snapshot 1`] = `
<div
  style={
    Object {
      "display": "block",
    }
  }
>
  <div
    className="popup-bg popup-open-bg"
  />
  <Styled(Card)
    className="popup-card-oh popup-open"
    extra={
      <img
        className="cardpopup-close"
        onClick={[MockFunction]}
        src={null}
      />
    }
    noBodymargin={true}
    style={
      Object {
        "overflow": "visible !important",
      }
    }
    title={
      <styled.div>
        Order Detail
      </styled.div>
    }
    width="480px"
  >
    <styled.div
      noMargin={true}
      width="100%"
    >
      <div
        className="status-list-wrapper"
      >
        <ul
          style={
            Object {
              "maxHeight": 256,
              "overflowY": "scroll",
              "padding": "0 10px",
            }
          }
        >
          <li>
            Sell 123 test at 123 undefined per test. Total 246.0 filled.
          </li>
          <li
            key="123"
            onClick={[Function]}
            style={
              Object {
                "cursor": "default",
              }
            }
          >
            <span>
              1970-01-01 00:00:00
            </span>
            <span>
              Invalid order
            </span>
          </li>
        </ul>
      </div>
    </styled.div>
  </Styled(Card)>
</div>
`;
